Src/DirectXHelpers.cpp

```diff
@@ -93,3 +93,170 @@ void DirectX::CreateShaderResourceView(
 
     device->CreateShaderResourceView(tex, &srvDesc, srvDescriptor);
 }
+
+_Use_decl_annotations_
+void DirectX::CreateUnorderedAccessView(
+    ID3D12Device* device,
+    ID3D12Resource* tex,
+    D3D12_CPU_DESCRIPTOR_HANDLE uavDescriptor,
+    UINT mipLevel)
+{
+    const auto desc = tex->GetDesc();
+
+    D3D12_UNORDERED_ACCESS_VIEW_DESC uavDesc = {};
+    uavDesc.Format = desc.Format;
+
+    switch (desc.Dimension)
+    {
+    case D3D12_RESOURCE_DIMENSION_TEXTURE1D:
+        if (desc.DepthOrArraySize > 1)
+        {
+            uavDesc.ViewDimension = D3D12_UAV_DIMENSION_TEXTURE1DARRAY;
+            uavDesc.Texture1DArray.MipSlice = mipLevel;
+            uavDesc.Texture1DArray.FirstArraySlice = 0;
+            uavDesc.Texture1DArray.ArraySize = desc.DepthOrArraySize;
+        }
+        else
+        {
+            uavDesc.ViewDimension = D3D12_UAV_DIMENSION_TEXTURE1D;
+            uavDesc.Texture1D.MipSlice = mipLevel;
+        }
+        break;
+
+    case D3D12_RESOURCE_DIMENSION_TEXTURE2D:
+        if (desc.DepthOrArraySize > 1)
+        {
+            uavDesc.ViewDimension = D3D12_UAV_DIMENSION_TEXTURE2DARRAY;
+            uavDesc.Texture2DArray.MipSlice = mipLevel;
+            uavDesc.Texture2DArray.ArraySize = desc.DepthOrArraySize;
+        }
+        else
+        {
+            uavDesc.ViewDimension = D3D12_UAV_DIMENSION_TEXTURE2D;
+            uavDesc.Texture2D.MipSlice = mipLevel;
+        }
+        break;
+
+    case D3D12_RESOURCE_DIMENSION_TEXTURE3D:
+        uavDesc.ViewDimension = D3D12_UAV_DIMENSION_TEXTURE3D;
+        uavDesc.Texture3D.MipSlice = mipLevel;
+        uavDesc.Texture3D.WSize = desc.DepthOrArraySize;
+        break;
+
+    case D3D12_RESOURCE_DIMENSION_BUFFER:
+        DebugTrace("ERROR: CreateUnorderedResourceView cannot be used with DIMENSION_BUFFER.\n\tUse CreateBufferUnorderedAccessView.");
+        throw std::exception("buffer resources not supported");
```

Comment:
The ``std::exception`` ctor that takes a string is a MSVC extension. Therefore, you should use one of the ``stdexcept`` types instead which by the C++ Standard can take strings as part of the ctor like I do elsewhere in the library.  ``std::runtime_error`` or ``std::invalid_argument`` ?